components/server-side-render/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies.
+ */
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Component,
+	RawHTML,
+} from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+export class ServerSideRender extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			response: null,
+		};
+	}
+
+	componentDidMount() {
+		this.fetch( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! isEqual( nextProps, this.props ) ) {
+			this.fetch( nextProps );
+		}
+	}
+
+	fetch( props ) {
+		this.setState( { response: null } );
+		const { block } = props;
+		const attributes = Object.assign( {}, props );
```

Comment:
Minor, but Gutenberg takes advantage of ES6 object spread syntax:

```js
const attributes = { ...props };
```

Reply:
See also https://github.com/WordPress/gutenberg/pull/5602#discussion_r175137283